contracts/collections/LinkedList.sol

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright ©️ 2018 Galt•Space Society Construction and Terraforming Company
+ * (Founded by [Nikolai Popeka](https://github.com/npopeka),
+ * [Dima Starodubcev](https://github.com/xhipster),
+ * [Valery Litvin](https://github.com/litvintech) by
+ * [Basic Agreement](http://cyb.ai/QmSAWEG5u5aSsUyMNYuX2A2Eaz4kEuoYWUkVBRdmu9qmct:ipfs)).
+ *
+ * Copyright ©️ 2018 Galt•Core Blockchain Company
+ * (Founded by [Nikolai Popeka](https://github.com/npopeka) and
+ * Galt•Space Society Construction and Terraforming Company by
+ * [Basic Agreement](http://cyb.ai/QmaCiXUmSrP16Gz8Jdzq6AJESY1EAANmmwha15uR3c1bsS:ipfs)).
+ */
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+library LinkedList {
+  struct Data {
+    mapping(uint256 => Node) nodesByIds;
+    uint256 headId;
+    uint256 count;
+  }
+
+  struct Node {
+    uint256 prevId;
+    uint256 nextId;
+  }
+
+  function insertByFoundAndComparator(Data storage data, uint256 newId, uint256 foundId, int8 compareResult) public {
+    if (data.headId == 0) {
+      data.count += 1;
+      
+      data.headId = newId;
+      return;
+    }
+
+    if (compareResult == 0) {
+      return;
+    } else if (compareResult < 0) {
+      // console.log('insertAfter', foundLeft);
+      insertAfter(data, newId, foundId);
+    } else {
+      if (foundId == data.headId) {
+        data.count += 1;
+
+        data.nodesByIds[newId].nextId = data.headId;
+
+        data.nodesByIds[data.headId].prevId = newId;
+        // console.log('insert head', newId, nodesByIds);
+        data.headId = newId;
+        return;
+      }
+      // console.log('headId', headId, nodesByIds[headId]);
+      // console.log('foundLeft', foundLeft, nodesByIds[foundLeft]);
```

Comment:
This is a definitely a wrong place for console.logs 😆

Reply:
😆